### DISCUSSION TITLE
Is it possible to avoid forced reflow？

### DISCUSSION THREAD

**Discussion by Alecyrus** [idiotWu/smooth-scrollbar#342] [Q&A] [answered]
## Environment
- Browser: latest chrome
- Version of smooth-scrollbar: latest

## Issue Summary
the call of getSize() function will cause forced reflow.

## Current Behavior

<img width="470" alt="截屏2019-10-0900 57 13" src="https://user-images.githubusercontent.com/13865581/66416132-c3273580-ea2f-11e9-9a00-6a470b1e98ef.png">
<img width="1135" alt="截屏2019-10-0900 58 23" src="https://user-images.githubusercontent.com/13865581/66416226-ed78f300-ea2f-11e9-8d6b-aac6ffdf81bb.png">
<img width="475" alt="截屏2019-10-0900 53 48" src="https://user-images.githubusercontent.com/13865581/66416078-a25ee000-ea2f-11e9-8609-70e777c0e143.png">

**Comment by Alecyrus** [comment]
@idiotWu  There is another way to get Size, please consider use `getBoundingClientRec` to obtain position and calculate the dimension of elements, which will definitely not lead to reflow.

**Comment by dolphin-wood** [comment, marked as answer]
`getBoundingClientRect` definitely forces layout/reflow too: https://gist.github.com/paulirish/5d52fb081b3570c81e3a
